apps/chat/bots.py

```diff
@@ -33,21 +33,23 @@ def notify_users_of_violation(session_id: int, safety_layer_id: int):
 
 
 class TopicBot:
-    def __init__(self, session: ExperimentSession):
-        experiment = session.experiment
-        self.prompt = experiment.prompt_text
-        self.input_formatter = experiment.input_formatter
-        self.llm = experiment.get_chat_model()
-        self.source_material = experiment.source_material.material if experiment.source_material else None
-        self.safety_layers = experiment.safety_layers.all()
+    def __init__(self, session: ExperimentSession, experiment: Experiment | None = None):
```

Comment:
a doc string explaining when to provide `experiment` would be useful

Reply:
https://github.com/dimagi/open-chat-studio/pull/432/commits/0b083492d44f1be3bdf9fefa5f3fb6469a70cf50 and https://github.com/dimagi/open-chat-studio/pull/432/commits/2a4ac2153d48b8ef404580c0c5397ef71d7d5225